### PR TITLE
Adds support to Verify emails, fixes `verify_delete` method

### PR DESCRIPTION
**Pull request by gabrielso**
1. Adds support to Verify emails
Added the VerifyEmailMessage entity
Adds example of Verify email

2. Sets version 3.0.0-rc.1

2. Fixes `verify_delete`
The client tries to instantiate a Verify object from the delete request but the response is empty.
Response body is nil and doesn't respond to `.empty?`.
Sending the request directly instead as other `*_delete` methods.

### CODE REVIEW

File: spec/verify_spec.rb [hunk 1]
```diff
@@ -39,4 +39,16 @@
 
     client.verify_create(31_612_345_678, originator: 'MessageBird')
   end
+
+  it 'creates a verify and sends token via email' do
```
Comment (kiriloman) [inline spec/verify_spec.rb:43]: Could you also add a test for `verify delete` please?

**Review by sbellity — APPROVED** [review]

File: messagebird-rest.gemspec [hunk 2]
```diff
@@ -9,7 +9,7 @@ Gem::Specification.new do |s|
   s.name        = 'messagebird-rest'
   s.version     = MessageBird::Version::STRING
   s.platform    = Gem::Platform::RUBY
-  s.required_ruby_version = '>= 1.9.3'
+  s.required_ruby_version = '>= 2.6.7'
```
[the remark this answers is not in the document]
Comment (gabrielso) [reply messagebird-rest.gemspec:12]: Would it be advised to allow versions that already reached end-of-life even releasing a new major version?

[the remark this answers is not in the document]
Comment (sbellity) [reply messagebird-rest.gemspec]: I would say that it's not really the responsibility of a library to enforce that. Ruby version requirements in gems are more about compatibility

**Review by kiriloman — APPROVED** [review]

File: lib/messagebird/verify.rb [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
+                  :messages,
+                  :status
+    attr_reader   :created_datetime,
+                  :valid_until_datetime
 
     def created_datetime=(value)
       @created_datetime = value_to_time(value)
     end
 
-    def valid_until_date_time=(value)
```
Comment (jalerson) [inline lib/messagebird/verify.rb:16]: Isn't a breaking change? Why it's necessary?

File: lib/messagebird/version.rb [hunk 4]
```diff
@@ -2,6 +2,6 @@
 
 module MessageBird
   module Version
-    STRING = '2.0.1'
```
Comment (jalerson) [inline lib/messagebird/version.rb:5]: This MR is only introducing new features to the gem, which is a minor upgrade. Why we're doing a major upgrade here?

File: spec/verify_spec.rb [hunk 5]
```diff
@@ -1,42 +1,56 @@
 # frozen_string_literal: true
 
 describe 'Verify' do
```
Comment (jalerson) [inline spec/verify_spec.rb:3]: I understand if you want to keep the test structure consistent with the other tests, however, rspec is such a powerful testing framework to write expressive tests, so I'd recommend:

* use `describe` to describe features
* use `context` to describe the feature in different situations/contexts
* use `it` to specify expectations (one `it` per `expect`)

For example:

```ruby

describe 'Verify an email OTP' do
  context 'using an existing OTP' do
    ...

    it 'has the ID verify-id' do
      expect(verify.id).to eq 'verify-id'
    end

    it 'has the verified status' do
      expect(verify.status).to eq 'verified'
    end
  end

  context 'using a non-existing OTP' do
    it 'raises an error' do
      expect { @client.verify('verify-id') }.to raise_error(MessageBird::ErrorException)
    end
  end

  context "<insert here any other scenario>" do
    ...
  end
end
```

**Review by jalerson — COMMENTED** [review]
Added a few comments and questions 👍

**Comment by jalerson** [comment]
Just found it's too late for my review 😅

File: lib/messagebird/verify.rb (hunk 3 above)
Comment (gabrielso) [reply lib/messagebird/verify.rb:16]: Hey @jalerson, the gem generates the attribute names based on the response from the API, and it converts `camelCase` to `snake_case`.

The response from the server contains the `validUntilDatetime` key, therefore the `valid_until_date_time` attribute breaks.

It is a breaking change that fixes a broken gem, and this is a major release as well, so I think it fits.

File: lib/messagebird/version.rb (hunk 4 above)
Comment (gabrielso) [reply lib/messagebird/version.rb:5]: See https://github.com/messagebird/ruby-rest-api/pull/62#discussion_r660814601

File: spec/verify_spec.rb (hunk 5 above)
Comment (gabrielso) [reply spec/verify_spec.rb:3]: Agreed, too late 😆 .

For the topmost `describe` block I usually prefer to use the class I'm testing, it gives me an instance of the class in the `subject` variable already to be used in the examples, and I can access it in some magic ways:

```ruby
describe Array do
  context "new instances" do
    it { should be_empty }
    # or
    it { is_expected.to be_empty }
  end
end
```

**Pull request merged**